src/server/project.ts

```diff
@@ -211,52 +214,162 @@ namespace ts.server {
             }
 
             this.setInternalCompilerOptionsForEmittingJsFiles();
+            const host = this.projectService.host;
+            if (host.trace) {
+                this.trace = s => host.trace(s);
+            }
 
-            this.lsHost = new LSHost(host, this, this.projectService.cancellationToken);
-            this.resolutionCache = createResolutionCache(
-                fileName => this.projectService.toPath(fileName),
-                () => this.compilerOptions,
-                directory => this.watchDirectoryOfFailedLookup(directory),
-                s => this.projectService.logger.info(s),
-                this.getProjectName(),
-                () => this.getTypeAcquisition().enable ? this.projectService.typingsInstaller.globalTypingsCacheLocation : undefined
-            );
-            this.lsHost.compilationSettings = this.compilerOptions;
-            this.resolutionCache.setModuleResolutionHost(this.lsHost);
-
-            this.languageService = createLanguageService(this.lsHost, this.documentRegistry);
+            if (host.realpath) {
+                this.realpath = path => host.realpath(path);
+            }
 
+            this.languageService = createLanguageService(this, this.documentRegistry);
             if (!languageServiceEnabled) {
                 this.disableLanguageService();
             }
-
+            this.resolutionCache = createResolutionCache(this);
             this.markAsDirty();
         }
 
-        private watchDirectoryOfFailedLookup(directory: string) {
+        getCompilationSettings() {
+            return this.compilerOptions;
+        }
+
+        getNewLine() {
+            return this.partialSystem.newLine;
+        }
+
+        getProjectVersion() {
+            return this.projectStateVersion.toString();
+        }
+
+        getScriptFileNames() {
+            const result: string[] = [];
+            if (this.rootFiles) {
+                this.rootFilesMap.forEach((value, _path) => {
+                    const f: ScriptInfo = isScriptInfo(value) && value;
+                    if (this.languageServiceEnabled || (f && f.isScriptOpen())) {
+                        // if language service is disabled - process only files that are open
+                        result.push(f ? f.fileName : value as NormalizedPath);

```

Comment:
I think this would be more readable as `isScriptInfo(value) && value.isScriptOpen()` and `isScriptInfo(value) ? value.fileName : value`, which doesn't need any casts, and ditch the variable `f`.
But if the language service isn't enabled, why would we have a bunch of scripts open?

Reply:
Language service is disabled when project exceeeds the size of allowed.. It still contains files open